pkg/diagnostics/collectors.go

```diff
@@ -120,62 +184,51 @@ func (c *collectorFactory) DefaultCollectors() []*Collect {
 	}
 }
 
-func (c *collectorFactory) EksaHostCollectors(machineConfigs []providers.MachineConfig) []*Collect {
-	var collectors []*Collect
-	collectorsMap := c.getCollectorsMap()
-
-	// we don't want to duplicate the collectors if multiple machine configs have the same OS family
-	osFamiliesSeen := map[v1alpha1.OSFamily]bool{}
-	for _, config := range machineConfigs {
-		if _, seen := osFamiliesSeen[config.OSFamily()]; !seen {
-			collectors = append(collectors, collectorsMap[config.OSFamily()]...)
-			osFamiliesSeen[config.OSFamily()] = true
-		}
+func (c *collectorFactory) defaultCrdCollectors() []*Collect {
+	defaultCrds := []string{
+		"clusters.anywhere.eks.amazonaws.com",
+		"bundles.anywhere.eks.amazonaws.com",
+		"clusters.cluster.x-k8s.io",
+		"machinedeployments.cluster.x-k8s.io",
+		"machines.cluster.x-k8s.io",
 	}
-	return collectors
+	return c.generateCrdCollectors(defaultCrds)
 }
 
-func (c *collectorFactory) getCollectorsMap() map[v1alpha1.OSFamily][]*Collect {
-	return map[v1alpha1.OSFamily][]*Collect{
-		v1alpha1.Ubuntu:       c.ubuntuHostCollectors(),
-		v1alpha1.Bottlerocket: c.modelRocketHostCollectors(),
+func (c *collectorFactory) generateCrdCollectors(crds []string) []*Collect {
+	var crdCollectors []*Collect
+	for _, d := range crds {
+		crdCollectors = append(crdCollectors, c.crdCollector(d))
 	}
+	return crdCollectors
 }
 
-func (c *collectorFactory) modelRocketHostCollectors() []*Collect {
-	return []*Collect{}
-}
-
-func (c *collectorFactory) ubuntuHostCollectors() []*Collect {
-	return []*Collect{
-		{
-			CopyFromHost: &copyFromHost{
-				Name:      "CloudInitLog",
-				Namespace: constants.EksaSystemNamespace,
-				Image:     c.DiagnosticCollectorImage,
-				HostPath:  "/var/log/cloud-init.log",
-			},
-		},
-		{
-			CopyFromHost: &copyFromHost{
-				Name:      "CloudInitOutputLog",
-				Namespace: constants.EksaSystemNamespace,
-				Image:     c.DiagnosticCollectorImage,
-				HostPath:  "/var/log/cloud-init-output.log",
-			},
-		},
-		{
-			CopyFromHost: &copyFromHost{
-				Name:      "Syslog",
-				Namespace: constants.EksaSystemNamespace,
-				Image:     c.DiagnosticCollectorImage,
-				HostPath:  "/var/log/syslog",
-				Timeout:   time.Minute.String(),
-			},
+func (c *collectorFactory) crdCollector(crdType string) *Collect {
+	command := []string{"kubectl"}
+	args := []string{"get", crdType, "-o", "json", "--all-namespaces"}
```

Comment:
curious why in all namespaces? Don't we know already what are the namespaces where we will find each CRD?

I personally don't think it's a big deal, buy that would allow us to scope down the permissions for the service account to get and list only on specific namespaces

Reply:
yeah I thought the increased visibility helped to ensure we wouldn't miss something 'out-of-place' by only focusing on the intended namespaces; for example if somehow one of our resources ended up in the wrong namespace, or a customer added a conflicting resource in another namespace, it'd be interesting to see it